lib/ansible/modules/cloud/amazon/elasticsearch.py

```diff
@@ -0,0 +1,407 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Ansible Project
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: elasticsearch
+short_description: Manage hosted Elasticsearch in Amazon Web Services.
+description:
+  - Manage hosted Elasticsearch in Amazon Web Services.
+  - Returns information about the hosted Elasticsearch.
+version_added: "2.5"
+requirements: [ boto3 ]
+author: "Zach Steindler (@steiza)"
+options:
+  domain:
+    description:
+      - Domain name for cluster.
+    required: true
+  state:
+    description:
+      - Create or destroy cluster
+    choices: [ "present", "absent" ]
+    default: present
+    required: false
+  version:
+    description:
+      - Version of Elasticsearch to use.
+    required: false
+    default: 5.1
+  instance_count:
+    description:
+      - Number of non-master nodes in the cluster.
+    required: false
+    default: 1
+  instance_type:
+    description:
+      - Instance type of non-master nodes in the cluster.
+    required: false
+    default: t2.medium.elasticsearch
```

Comment:
I wouldn't default parameters that are passed to AWS.  It leads to unexpected consequences.

For example, imagine someone used your module to start managing a cluster they'd already created with a m3.medium type.  They didn't specify the `instance_type` parameter.  When they run the play, your module *might* (i haven't checked your code) change the type to a t2.medium which is not what was wanted